src/scorecard.js

```diff
@@ -0,0 +1,73 @@
+var Scorecard = function(content) {
+  this.frameRecord = [];
+  this.totals = [];
+  this.DEFAULT_FRAMES = 10;
+  this.currentTurn = 0;
+  this.maxTurns = 20;
+  for (var i = 0; i < this.DEFAULT_FRAMES; i++) {
+    this.frameRecord.push(new content);
+  };
+};
+
+Scorecard.prototype.updateTotal = function() {
+  if (this.frameRecord[this.getCurrentFrame()].strike === true || this.frameRecord[this.getCurrentFrame()].spare === true) {
+    this.totals.push(10);
+  } else {
+    this.totals.push(this.frameRecord[this.getCurrentFrame()].result);  
+  };
+};
+
+Scorecard.prototype.checkLastFrameForStrike = function() {
+  if (this.getCurrentFrame() > 0 && this.frameRecord[this.getCurrentFrame() - 1].strike === true) return true;
+};
+
+Scorecard.prototype.checkLastFrameForSpare = function() {
+  if (this.getCurrentFrame() > 0 && this.frameRecord[this.getCurrentFrame() - 1].spare === true && this.frameRecord[this.getCurrentFrame()].firstBowl === null) return true;
+}
+
+Scorecard.prototype.checkTwoLastFramesForStrike = function() {
+  if (this.getCurrentFrame() > 1 && this.frameRecord[this.getCurrentFrame() - 2].strike === true && this.frameRecord[this.getCurrentFrame()].firstBowl === null) return true;
+}
+
+Scorecard.prototype.checkCurrentFrameForStrikeWith = function(pinsKnockedOver) {
+  if (this.checkEven(this.currentTurn) === true && pinsKnockedOver === 10) this.currentTurn +=1;
+}
+
+Scorecard.prototype.checkIfFrameIsOver = function() {
+  if (this.checkEven(this.currentTurn) != true) this.updateTotal();
+}
+
+Scorecard.prototype.bowl = function(pinsKnockedOver) {
+  if (this.currentTurn === 20 && this.totals[this.getCurrentFrame()] === 10) {
+    this.totals[this.getCurrentFrame()] += pinsKnockedOver;
+    if (this.checkLastFrameForStrike() === true) this.totals[this.getCurrentFrame() - 1] += pinsKnockedOver;
+    this.currentTurn ++
+  } else if (this.currentTurn === 21 && this.frameRecord[this.getCurrentFrame()].strike === true) {
+    this.totals[this.getCurrentFrame()] += pinsKnockedOver;
+    this.currentTurn ++
+  } else if (this.currentTurn < this.maxTurns) {
+    this.checkCurrentFrameForStrikeWith(pinsKnockedOver);
+    if (this.checkLastFrameForStrike() === true) {
+        this.totals[this.getCurrentFrame() - 1] += pinsKnockedOver;
+        if (this.checkTwoLastFramesForStrike() === true) this.totals[this.getCurrentFrame() - 2] += pinsKnockedOver;
+    }
+    if (this.checkLastFrameForSpare() === true) this.totals[this.getCurrentFrame() - 1] += pinsKnockedOver;
+    this.frameRecord[this.getCurrentFrame()].receiveBowl(pinsKnockedOver);
+    this.checkIfFrameIsOver();
+    this.currentTurn += 1;
+  };
+};
+
+Scorecard.prototype.getCurrentFrame = function() {
+  if (this.currentTurn >= 20) return 9;
```

Comment:
Calculating the current frame by checking if the current turn is even then subtracting 1 and dividing by 2 is highly brittle.  Why not just track the current frame separately?  This kind of anti-pattern quickly leads to problems where one variable is shoehorned into 2 responsibilities.